src/test/java/org/folio/rest/impl/PatronNoticeResourceImplTest.java

```diff
@@ -0,0 +1,158 @@
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.http.HttpHeaders;
+import org.awaitility.Awaitility;
+import org.folio.client.OkapiModulesClient;
+import org.folio.helper.OkapiModulesClientHelper;
+import org.folio.rest.jaxrs.model.Context;
+import org.folio.rest.jaxrs.model.Errors;
+import org.folio.rest.jaxrs.model.NotifySendRequest;
+import org.folio.rest.jaxrs.model.PatronNoticeEntity;
+import org.folio.rest.jaxrs.model.TemplateProcessingRequest;
+import org.folio.rest.jaxrs.model.TemplateProcessingResult;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PatronNoticeResourceImplTest {
+  private static final String LANG = "lang";
+
+  @Spy
+  private PatronNoticeResourceImpl patronNoticeResource;
+
+  @Mock
+  private OkapiModulesClient client;
+
+  @Spy
+  private OkapiModulesClientHelper okapiModulesClientHelper;
+
+  private Map<String, String> okapiHeaders = new HashMap<>();
+  private PatronNoticeEntity entity = makeEntity();
+  private TemplateProcessingResult templateProcessingResult = new TemplateProcessingResult();
+  private TemplateProcessingRequest templateProcessingRequest = new TemplateProcessingRequest();
+
+  private boolean handlerIsCalled;
+
+  @Before
+  public void setUp() {
+    doReturn(client).when(patronNoticeResource).makeOkapiModulesClient(any(), any());
+
+    doReturn(succeededFuture(templateProcessingResult)).when(client)
+      .postTemplateRequest(templateProcessingRequest);
+
+    doReturn(templateProcessingRequest).when(okapiModulesClientHelper)
+      .buildTemplateProcessingRequest(entity);
+
+    doReturn(new NotifySendRequest()).when(okapiModulesClientHelper)
+      .buildNotifySendRequest(templateProcessingResult, entity);
+
+    doReturn(okapiModulesClientHelper).when(patronNoticeResource).getOkapiModulesClientHelper();
+
+    handlerIsCalled = false;
+  }
+
+  @Test
+  public void shouldCallHandlerWith200StatusWhenMessageIsDelivered() {
+    Handler<AsyncResult<Response>> handler = responseAsyncResult -> {
+      handlerIsCalled = true;
+
+      assertTrue(responseAsyncResult.succeeded());
+      assertEquals(200, responseAsyncResult.result().getStatus());
+
+      verify(okapiModulesClientHelper, times(1)).buildTemplateProcessingRequest(entity);
+      verify(okapiModulesClientHelper, times(1)).buildNotifySendRequest(
```

Comment:
Second argument of `verify()` can be omitted when you expect an event to happen only once.

Reply:
I think verify() without times(1) just checks that there was at least one interaction rather than exactly one.

Reply:
```
     * Verifies certain behavior <b>happened once</b>.
     * <p>
     * Alias to <code>verify(mock, times(1))</code> E.g:
```

Reply:
Good to know, thanks.